packages/store/src/state-factory.ts

```diff
@@ -117,7 +117,7 @@ export class StateFactory {
           if (result instanceof Observable) {
             result = result.pipe(
               (<ActionOptions>actionMeta.options).cancelUncompleted
-                ? takeUntil(actions$.pipe(ofAction(action.constructor)))
+                ? takeUntil(actions$.pipe(ofActionDispatched(action.constructor)))
```

Comment:
is `action` here the class or instance? It looks like instance and if so, this will cause issues if the action dispatched is not a class (think devtools or websocket). We probably need to change the `action.constructor` to check and return.

Reply:
it is counting on an instance yes. we can make this more robust but we should also be safer in the operators as well

Reply:
actually I should probably put all of that logic in the operators anyway